go-controller/pkg/config/config.go

```diff
@@ -652,6 +653,35 @@ func InitConfigWithPath(ctx *cli.Context, exec kexec.Interface, configFile strin
 	overrideFields(&CNI, &cfg.CNI)
 	overrideFields(&CNI, &cliConfig.CNI)
 
+	// values for token, cacert, kubeconfig, api-server may be found in several places.
+	// Take the first found when looking in this order: command line options, environment variables, files
+
+	if cliConfig.Kubernetes.Token == "" {
+		cliConfig.Kubernetes.Token = os.Getenv("K8S_TOKEN")
+	}
+	if cliConfig.Kubernetes.Token == "" {
+		if data, err1 := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token"); err1 == nil {
+			cliConfig.Kubernetes.Token = string(data)
+		}
+	}
+
+	if cliConfig.Kubernetes.CACert == "" {
+		cliConfig.Kubernetes.CACert = os.Getenv("K8S_CACERT")
+	}
+	if cliConfig.Kubernetes.CACert == "" {
+		if _, err2 := os.Stat("/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"); err2 == nil {
+			cliConfig.Kubernetes.CACert = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+		}
+	}
+
+	if cliConfig.Kubernetes.Kubeconfig == "" {
+		cliConfig.Kubernetes.Kubeconfig = os.Getenv("KUBECONFIG")
+	}
+
+	if cliConfig.Kubernetes.APIServer == "" {
+		cliConfig.Kubernetes.APIServer = os.Getenv("K8S_APISERVER")
+	}
+
```

Comment:
@pecameron can we do this in buildKubernetesConfig() instead of in InitConfigWithPath()?

Reply:
OK. Wasn't sure about guidelines on what went where.